tests/util/test_full_block_utils.py

```diff
@@ -0,0 +1,219 @@
+    fees = uint64(random.randint(0, 150000))
+
+    yield TransactionsInfo(
+        rand_hash(),  # generator_root
+        rand_hash(),  # generator_refs_root
+        rand_g2(),  # aggregated_signature
+        fees,
+        uint64(random.randint(0, 12000000000)),  # cost
+        reward_claims_incorporated,
+    )
+
+
+def get_challenge_chain_sub_slot():
+    for infused_chain_sub_slot_hash in [rand_hash(), None]:
+        for sub_epoch_summary_hash in [rand_hash(), None]:
+            for new_sub_slot_iters in [uint64(random.randint(0, 4000000000)), None]:
+                for new_difficulty in [uint64(random.randint(1, 30)), None]:
+                    yield ChallengeChainSubSlot(
+                        rand_vdf(),  # challenge_chain_end_of_slot_vdf
+                        infused_chain_sub_slot_hash,
+                        sub_epoch_summary_hash,
+                        new_sub_slot_iters,
+                        new_difficulty,
+                    )
+
+
+def get_reward_chain_sub_slot():
+    for infused_challenge_chain_sub_slot_hash in [rand_hash(), None]:
+        yield RewardChainSubSlot(
+            rand_vdf(),  # end_of_slot_vdf
+            rand_hash(),  # challenge_chain_sub_slot_hash
+            infused_challenge_chain_sub_slot_hash,
+            uint8(random.randint(0, 255)),  # deficit
+        )
+
+
+def get_sub_slot_proofs():
+    for infused_challenge_chain_slot_proof in [rand_vdf_proof(), None]:
+        yield SubSlotProofs(
+            rand_vdf_proof(),  # challenge_chain_slot_proof
+            infused_challenge_chain_slot_proof,
+            rand_vdf_proof(),  # reward_chain_slot_proof
+        )
+
+
+def get_end_of_sub_slot():
+    for challenge_chain in get_challenge_chain_sub_slot():
+        for infused_challenge_chain in [InfusedChallengeChainSubSlot(rand_vdf()), None]:
+            for reward_chain in get_reward_chain_sub_slot():
+                for proofs in get_sub_slot_proofs():
+                    yield EndOfSubSlotBundle(
+                        challenge_chain,
+                        infused_challenge_chain,
+                        reward_chain,
+                        proofs,
+                    )
+
+
+def get_finished_sub_slots():
+    yield []
+    yield [s for s in get_end_of_sub_slot()]
+
+
+def get_full_blocks():
+
+    random.seed(123456789)
+
+    generator = SerializedProgram.from_bytes(bytes.fromhex("ff01820539"))
+
+    for foliage in get_foliage():
+        for foliage_transaction_block in get_foliage_transaction_block():
+            height = random.randint(0, 1000000)
+            for reward_chain_block in get_reward_chain_block(height):
+                for transactions_info in get_transactions_info(height):
+                    for challenge_chain_sp_proof in [rand_vdf_proof(), None]:
+                        for reward_chain_sp_proof in [rand_vdf_proof(), None]:
+                            for infused_challenge_chain_ip_proof in [rand_vdf_proof(), None]:
```

Comment:
Is it important to the parser whether the block is partly or fully compactified?

eg for `challenge_chain_sp_proof` and `infused_challenge_chain_ip_proof`

Reply:
no, because the proof itself is just `bytes`, so it's handled correctly regardless of length

Reply:
right, I just wanted the test to be somewhat exhaustive. There are variable length fields that aren't exercised at very many different lengths, but at least all combinations of optionals are (most of which are invalid blocks anyway). It takes about 4-5 mintues to run through all of these cases.